activesupport/lib/active_support/duration/iso8601_serializer.rb

```diff
@@ -40,7 +41,7 @@ def normalize
           end
           # If all parts are negative - let's make a negative duration
           sign = ''
-          if parts.values.all? { |v| v < 0 }
+          if !parts.empty? && parts.values.all? { |v| v < 0 }
```

Comment:
As I mentioned in the last paragraph of the PR description, `{}.values.all? #=> true`. I made the change here because it would be nice if `#normalize` didn't return sign `-` if the value of the duration is `0`, even if the current implementation of `#serialize` doesn't use it.